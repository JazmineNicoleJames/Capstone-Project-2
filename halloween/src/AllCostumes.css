
.list-group a {
    margin-top: 1rem;
    color: white;
    padding: 1px;
    text-decoration: none;
}

.list-group a:not(:last-of-type) li {
    border-bottom: none !important;
}

.list-group a:hover {
    text-decoration: none;
    border-left-style: solid;
    border-left-width: 3px;
}

.list-group a:hover li {
    margin-left: 2px;
}

.list-group .list-inline {
    background-color: transparent;
    padding: 5px;
    text-transform: capitalize;
}

.list-group .list-inline:hover {
    cursor: pointer;
}
