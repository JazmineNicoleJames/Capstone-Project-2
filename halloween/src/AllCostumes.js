import React from "react";
import { Link } from "react-router-dom";
import {ListGroup, List, ListGroupItem} from "reactstrap";
import "./AllCostumes.css";

/* AllCostumes component

    * allCostumes prop passed down from  App.js

    **    returns a list of all costumes - listed by PK item_name

** or "No costumes available"

*/
const AllCostumes = ({allCostumes}) => {

    return (
        <div name="AllCostumes">
            {allCostumes && allCostumes.length > 0 ? (
                    <ListGroup>
               {allCostumes.map(costume => (
                <List type='inline' key={costume.item_name}>
                    <Link to={{pathname: `/costumes/${costume.item_name}`, 
                    state: {allCostumes: costume}}}>
                {costume.item_name.includes("-")
                  ? costume.item_name.replace(/-/g, " ")
                  : costume.item_name}
                    </Link>
                </List>
            ))}
            </ListGroup> 

    ) : ( 
        <p> No costumes available</p>
    )}
        </div>
    )
};

export default AllCostumes;