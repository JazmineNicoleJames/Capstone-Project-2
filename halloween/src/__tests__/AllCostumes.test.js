import React from "react";
import { MemoryRouter } from "react-router-dom";
import { fireEvent, render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import AllCostumes from '../AllCostumes';

jest.mock('../App', () => ({
    __esModule: true,
    default:() => ({
        allCostumes: [
            {item_name: 'Grim Reaper'}
        ]
    })
}))

test("should match snapshot", () => {
    const allCostumes = [{item_name: 'Grim Reaper'}]
    const { asFragment } = render(
        <MemoryRouter>
            <AllCostumes allCostumes={allCostumes}/>
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
});

test("renders AllCostumes", () => {
    const allCostumes = [{item_name: 'Grim Reaper'}]
    const { getByText } = render(
        <MemoryRouter>
            <AllCostumes allCostumes={allCostumes} />
        </MemoryRouter>
    );
    expect(getByText("Grim Reaper")).toBeInTheDocument();
});

test('fails to renders AllCostumes, displays msg', () => {
    const { getByText } = render(
        <MemoryRouter>
            <AllCostumes />
        </MemoryRouter>
    );
    
    expect(getByText("No costumes available")).toBeInTheDocument();
});


