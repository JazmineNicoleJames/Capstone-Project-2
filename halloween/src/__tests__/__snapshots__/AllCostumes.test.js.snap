// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="list-group"
    >
      <ul
        class="list-inline"
      >
        <a
          href="/costumes/Grim Reaper"
        >
          Grim Reaper
        </a>
      </ul>
    </ul>
  </div>
</DocumentFragment>
`;
